bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieNettyServer.java

```diff
@@ -302,6 +303,9 @@ protected void initChannel(SocketChannel ch) throws Exception {
                         new BookieSideConnectionPeerContextHandler();
                     ChannelPipeline pipeline = ch.pipeline();
 
+                    // For ByteBufList, skip the usual LengthFieldPrepender and have the encoder itself to add it
+                    pipeline.addLast("bytebufList", ByteBufList.ENCODER_WITH_SIZE);
```

Comment:
If I understand correctly, we are not changing the structure on the wire, it is 100% compatible with old clients, can you confirm @merlimat ?

Reply:
Yes, there is absolutely no change in the wire format. The "encoder" here is to translate the `ByteBufList` into a series of `ByteBuf` to be written on the socket. 

The `ENCODER_WITH_SIZE` is to avoid the regular `LenghtFieldPrepender` to insert the size header for each item of the list. Since the `ByteBufList.Encoder` is the last encoder in the sequence, `LenghtFieldPrepender` will ignore the `ByteBufList` and we're manually adding the size (only once) in the encoder itself.